src/views/cross_reference.ts

```diff
@@ -53,17 +53,17 @@ export class CrossReferenceView
   }
 
   public updateContents = () => {
-    const targets = objectsKey.getState(this.view.state) as Map<string, Target>
+    window.setTimeout(() => {
+      const targets = objectsKey.getState(this.view.state) as Map<string, Target>
+      const attrs = this.node.attrs
+      const classes = ['cross-reference', ...getChangeClasses(attrs.dataTracked)]
+      this.dom.className = classes.join(' ')
 
-    const attrs = this.node.attrs
-
-    const classes = ['cross-reference', ...getChangeClasses(attrs.dataTracked)]
-
-    this.dom.className = classes.join(' ')
-    const label = attrs.rids.length && targets.get(attrs.rids[0])?.label
-
-    this.dom.textContent = attrs.label || label || ''
-    this.dom.addEventListener('click', this.handleClick)
+      const label = attrs.rids.length && targets.get(attrs.rids[0])?.label
+      // attrs.label contains custom text inserted at cross-reference creation time
+      this.dom.textContent = attrs.label || label || ''
+      this.dom.addEventListener('click', this.handleClick)
+    }, 0)
```

Comment:
we need to replace `attrs.label || label` to use plugin label first `label || attrs.label`. and we don't need `window.setTimeout` or any other change in this file as updating node decoration will retrigger `updateContents`